File: GZCTF/Models/Request/Admin/BasicUserInfoModel.cs
namespace CTFServer.Models.Request.Admin;

public class BasicUserInfoModel
{
    /// <summary>
    /// 用户ID
    /// </summary>
    public string? Id { get; set; }

    /// <summary>
    /// 用户名
    /// </summary>
    public string? UserName { get; set; }

    /// <summary>
    /// 邮箱
    /// </summary>
    public string? Email { get; set; }

    /// <summary>
    /// 头像链接
    /// </summary>
    public string? Avatar { get; set; }

    /// <summary>
    /// 用户角色
    /// </summary>
    public Role? Role { get; set; }

    /// <summary>
    /// 所拥有的队伍
    /// </summary>
    public string? OwnTeamName { get; set; }

    public int? OwnTeamId { get; set; }

    /// <summary>
    /// 激活的队伍
    /// </summary>
    public string? ActiveTeamName { get; set; }

    public int? ActiveTeamId { get; set; }

    public static BasicUserInfoModel FromUserInfo(UserInfo user)
    => new()
    {
        Id = user.Id,
        UserName = user.UserName,
        Email = user.Email,
        Avatar = user.AvatarUrl,
        Role = user.Role,
        OwnTeamId = user.OwnTeamId,
        OwnTeamName = user.OwnTeam?.Name,
        ActiveTeamId = user.ActiveTeamId,
        ActiveTeamName = user.ActiveTeam?.Name
    };
}

File: GZCTF/ClientApp/pages/client/models/BasicUserInfoModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Role } from './Role';

export type BasicUserInfoModel = {
    /**
     * 用户ID
     */
    id?: string | null;
    /**
     * 用户名
     */
    userName?: string | null;
    /**
     * 邮箱
     */
    email?: string | null;
    /**
     * 头像链接
     */
    avatar?: string | null;
    /**
     * 用户角色
     */
    role?: Role | null;
    /**
     * 所拥有的队伍
     */
    ownTeamName?: string | null;
    ownTeamId?: number | null;
    /**
     * 激活的队伍
     */
    activeTeamName?: string | null;
    activeTeamId?: number | null;
};
